includes/abstracts/abstract-wc-payment-gateway-payex.php

```diff
@@ -165,6 +214,13 @@ public function request( $method, $url, $params = array() ) {
 				}
 			}
 
+			// Message for invalid Msisdn
+			if ( ( strpos( $message, 'The Msisdn is not valid' ) !== false ) ||
```

Comment:
Can't you use something else than `message` for this? Isn't `type` or `problems` detailed enough to extract this sort of information?

Reply:
How does the entire `application/problem+json` look like for this validation error? Can't you deserialize the `$message` into an object and look at its `type` or `problems` properties to figure out this in a more programmatic-friendly way than searching through a string after a human readable error message?

Reply:
This is still unresolved. Treat `problem+json` responses as structured JSON objects, not as a string.